### PR TITLE
Added Audience field in duckv1.Addressable

### LINKED ISSUE #2795: Add `Audience` field in `duckv1.Addressable`
To allow Addressables to provide their Audience field, the `duckv1.Addressable` needs to be extended to include the `Audience`:

```
type Addressable struct {
    
    ...

	// Audience is the OIDC audience for this address.
	// +optional
	Audience *string `json:"audience,omitempty"`
}

```

**Time Estimate (optional)**

1 

**Additional context (optional)**

/good-first-issue

### PR DESCRIPTION
**Pull request by abhijeetgauravm**
# changes

- Added Audience field in duckv1.Addressable

/kind <kind>

Fixes #2795 

**Release Note**

```release-note

```

**Docs**

```docs

```

### CODE REVIEW

**Comment by creydr** [comment]
Hello @abhijeetgauravm,
Thanks for your contribution. 
Here applies the same as for https://github.com/knative/pkg/pull/2797#issuecomment-1681910427. Can you run [`hack/update-codegen.sh`](https://github.com/knative/pkg/blob/main/hack/update-codegen.sh) to generate/update the `DeepCopy` functions to include the new field?

**Comment by pierDipi** [comment]
@abhijeetgauravm any updates related to the comment above?

**Comment by abhijeetgauravm** [comment]
> @abhijeetgauravm any updates related to the comment above?

hey @pierDipi actually  should i run update-codegen.sh command on my terminal to generate/update the DeepCopy functions. Actually i am beginner . Could you help a bit?

**Comment by pierDipi** [comment]
@abhijeetgauravm yes, you need to run `hack/update-codegen.sh` in your terminal and then commit/push the difference

**Comment by abhijeetgauravm** [comment]
> hack/update-codegen.sh

image
but doing this terminal nothing happens in terminal. Should it push it only running the command like this.

**Comment by creydr** [comment]
> > hack/update-codegen.sh
> 
> image but doing this in terminal nothing happens in terminal. Should i push it only running the command like this @pierDipi

Hi @abhijeetgauravm,
the `hack/update-codegen.sh` is a shell/bash script, so you need to run it on some kind of UNIX machine or use something like WSL when you are on Windows. Also you should run it from the repo root dir AFAIK (`./hack/update-codegen.sh`)

**Comment by abhijeetgauravm** [comment]
@creydr done !

File: apis/duck/v1/addressable_types.go [hunk 1]
```diff
@@ -48,6 +48,11 @@ type Addressable struct {
 	// according to https://www.rfc-editor.org/rfc/rfc7468.
 	// +optional
 	CACerts *string `json:"CACerts,omitempty"`
+
+
```
Comment (creydr) [inline apis/duck/v1/addressable_types.go:52]:
```suggestion
```
@abhijeetgauravm can you remove one empty line?

**Review by creydr — COMMENTED** [review]
Thanks for your contribution @abhijeetgauravm. I left one comment about the code style

**Review by creydr — APPROVED** [review]
Thanks a lot for you contribution and patience @abhijeetgauravm 

/lgtm

**Pull request merged**